lib/settings.coffee

```diff
@@ -70,22 +70,21 @@ config = require('./config')
 
 readConfigFile = (file) ->
 	try
-
 		# We read the config files synchronously since
 		# other modules rely on Resin Settings Client
 		# to be ready for usage as soon as possible.
 		return yaml.parse(fs.readFileSync(file, encoding: 'utf8'))
-
 	catch error
 		return {} if error.code is 'ENOENT'
 		throw error
 
-settings = utils.mergeObjects.apply null, [
-	defaults
-	readConfigFile(config.paths.user)
-	readConfigFile(config.paths.project)
-	environment.parse(process.env)
-]
+getSettings = _.memoize ->
+	utils.mergeObjects.apply null, [
```

Comment:
Do you know if this can be:
```coffee
utils.mergeObjects(
	defaults
	readConfigFile(config.paths.user)
	readConfigFile(config.paths.project)
	environment.parse(process.env)
)
```
?